Automattic/bugomattic
# Add initial repository bootstrapping

## Changes

This PR establishes the initial, overall bootstrapping for the repo. Namely, it adds the basics of a React app, basic build scripts, and things like linting/formatting.

Notably, it does not add...
- Any meaningful functionality whatsoever - the actual app content is just boilerplate junk.
- Redux - we'll add it when we actually add the app state to the application
- GitHub templates - we'll do it later.

Overall, I've tried to strike the (very difficult!) balance between making standards & approaches match things we do elsewhere in A8c (like in Calypso), and also making things fit best practices for the size and scope of the app. 

## Testing

Clone the repo and checkout this branch. Then, play around!

- Run `yarn` to install dependencies.
- Run `yarn start` to see the development build.
- Run `yarn build` to see a prod build output.
- Run `yarn test` to see how tests run. 
- Play around in code files to see the linting.

Also, please let me know any feedback you have!
- Are you good with the linting/formatting approach and rules?
- Anything missing in the package.json and scripts?
- Any concerns with the big-picture decisions? e.g. typescript, css modules (vs css-in-js)
